Build failure: usbmuxd2 on unstable

## Steps To Reproduce

Steps to reproduce the behavior:

1. build *usbmuxd2* on the nixos-unstable or nixpkgs-unstable branch

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/9cdr4cm8b4sqsidcljvkkx5p27mrz8hl-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: autoreconfPhase
@nix { "action": "setPhase", "phase": "autoreconfPhase" }
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I m4
aclocal: warning: couldn't open directory 'm4': No such file or directory
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force -I m4
autoreconf: running: /nix/store/m0knd99ncm7kbpffm10gsdmgv1j6mgy6-autoconf-2.72/bin/autoconf --force
configure.ac:4: warning: The macro 'AC_CANONICAL_SYSTEM' is obsolete.
configure.ac:4: You should run autoupdate.
./lib/autoconf/general.m4:2081: AC_CANONICAL_SYSTEM is expanded from...
configure.ac:4: the top level
configure.ac:11: warning: The macro 'AM_PROG_LIBTOOL' is obsolete.
configure.ac:11: You should run autoupdate.
m4/libtool.m4:101: AM_PROG_LIBTOOL is expanded from...
configure.ac:11: the top level
autoreconf: running: /nix/store/m0knd99ncm7kbpffm10gsdmgv1j6mgy6-autoconf-2.72/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:11: installing './compile'
configure.ac:4: installing './config.guess'
configure.ac:4: installing './config.sub'
configure.ac:12: installing './install-sh'
configure.ac:12: installing './missing'
usbmuxd2/Makefile.am: installing './depcomp'
autoreconf: './install-sh' is updated
autoreconf: Leaving directory '.'
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing libtool script ./ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/81p3wwkrgxhhad2q60j9gb4i14ipg2iz-usbmuxd2-unstable-2023-12-12 --with-udevrulesdir=/nix/store/81p3wwkrgx>
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether clang accepts -g... yes
checking for clang option to enable C11 features... none needed
checking whether clang understands -c and -o together... yes
checking for a sed that does not truncate output... /nix/store/abm77lnrkrkb58z6xp1qwjcr1xgkcfwm-gnused-4.9/bin/sed
checking for grep that handles long lines and -e... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep
checking for egrep... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -E
checking for fgrep... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -F
checking for ld used by clang... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking for gawk... gawk
checking command to parse nm output from clang object... ok
checking for sysroot... no
checking for a working dd... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/dd
checking how to truncate binary pipes... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if clang supports -fno-rtti -fno-exceptions... yes
checking for clang option to produce PIC... -fPIC -DPIC
checking if clang PIC flag -fPIC -DPIC works... yes
checking if clang static flag -static works... no
checking if clang supports -c -o file.o... yes
checking if clang supports -c -o file.o... (cached) yes
checking whether the clang linker (ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for a BSD-compatible install... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/mkdir -p
checking whether make sets $(MAKE)... yes
checking whether make supports the include directive... yes (GNU style)
checking whether make supports nested variables... yes
checking dependency style of clang... none
checking whether the compiler supports GNU C++... yes
checking whether clang++ accepts -g... yes
checking for clang++ option to enable C++11 features... none needed
checking how to run the C++ preprocessor... clang++ -E
checking for ld used by clang++... ld -m elf_x86_64
checking if the linker (ld -m elf_x86_64) is GNU ld... yes
checking whether the clang++ linker (ld -m elf_x86_64) supports shared libraries... yes
checking for clang++ option to produce PIC... -fPIC -DPIC
checking if clang++ PIC flag -fPIC -DPIC works... yes
checking if clang++ static flag -static works... no
checking if clang++ supports -c -o file.o... yes
checking if clang++ supports -c -o file.o... (cached) yes
checking whether the clang++ linker (ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
autoreconf: running: /nix/store/m0knd99ncm7kbpffm10gsdmgv1j6mgy6-autoconf-2.72/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:11: installing './compile'
configure.ac:4: installing './config.guess'
configure.ac:4: installing './config.sub'
configure.ac:12: installing './install-sh'
configure.ac:12: installing './missing'
usbmuxd2/Makefile.am: installing './depcomp'
autoreconf: './install-sh' is updated
autoreconf: Leaving directory '.'
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing libtool script ./ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/81p3wwkrgxhhad2q60j9gb4i14ipg2iz-usbmuxd2-unstable-2023-12-12 --with-udevrulesdir=/nix/store/81p3wwkrgxhhad2q60j9gb4i14ipg2iz-usbmuxd2-unstable-2023-12>
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether clang accepts -g... yes
checking for clang option to enable C11 features... none needed
checking whether clang understands -c and -o together... yes
checking for a sed that does not truncate output... /nix/store/abm77lnrkrkb58z6xp1qwjcr1xgkcfwm-gnused-4.9/bin/sed
checking for grep that handles long lines and -e... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep
checking for egrep... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -E
checking for fgrep... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -F
checking for ld used by clang... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking for gawk... gawk
checking command to parse nm output from clang object... ok
checking for sysroot... no
checking for a working dd... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/dd
checking how to truncate binary pipes... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if clang supports -fno-rtti -fno-exceptions... yes
checking for clang option to produce PIC... -fPIC -DPIC
checking if clang PIC flag -fPIC -DPIC works... yes
checking if clang static flag -static works... no
checking if clang supports -c -o file.o... yes
checking if clang supports -c -o file.o... (cached) yes
checking whether the clang linker (ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for a BSD-compatible install... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/mkdir -p
checking whether make sets $(MAKE)... yes
checking whether make supports the include directive... yes (GNU style)
checking whether make supports nested variables... yes
checking dependency style of clang... none
checking whether the compiler supports GNU C++... yes
checking whether clang++ accepts -g... yes
checking for clang++ option to enable C++11 features... none needed
checking how to run the C++ preprocessor... clang++ -E
checking for ld used by clang++... ld -m elf_x86_64
checking if the linker (ld -m elf_x86_64) is GNU ld... yes
checking whether the clang++ linker (ld -m elf_x86_64) supports shared libraries... yes
checking for clang++ option to produce PIC... -fPIC -DPIC
checking if clang++ PIC flag -fPIC -DPIC works... yes
checking if clang++ static flag -static works... no
checking if clang++ supports -c -o file.o... yes
checking if clang++ supports -c -o file.o... (cached) yes
checking whether the clang++ linker (ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking dependency style of clang++... none
checking for gcc... (cached) clang
checking whether the compiler supports GNU C... (cached) yes
checking whether clang accepts -g... (cached) yes
checking for clang option to enable C11 features... (cached) none needed
checking whether clang understands -c and -o together... (cached) yes
checking whether we need platform-specific build settings... checking pkg-config is at least version 0.9.0... yes
checking for libgeneral... yes
checking for libusb-1.0 >= 1.0.9... yes
checking for libplist-2.0 >= 2.2.0... yes
checking for avahi-client >= 0.7... yes
checking for libimobiledevice-1.0 >= 1.3.0... yes
checking for struct sockaddr.sa_len... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating udev/Makefile
config.status: creating systemd/Makefile
config.status: creating usbmuxd2/Makefile
config.status: creating debian/usbmuxd2.control
config.status: creating config.h
config.status: executing libtool commands
config.status: executing depfiles commands

Configuration for usbmuxd2-1:
-------------------------------------------

  install prefix ..........: /nix/store/81p3wwkrgxhhad2q60j9gb4i14ipg2iz-usbmuxd2-unstable-2023-12-12
  Debug build .............: false
  preflight support .......: yes
  WIFI support ............: yes
  activation method .......: systemd
  WIFI backend ............: avahi
  systemd unit directory ..: /nix/store/81p3wwkrgxhhad2q60j9gb4i14ipg2iz-usbmuxd2-unstable-2023-12-12/lib/systemd/system
  udev rules directory ....: /nix/store/81p3wwkrgxhhad2q60j9gb4i14ipg2iz-usbmuxd2-unstable-2023-12-12/lib/udev/rules.d
  compiler ................: clang

  Now type 'make' to build usbmuxd2-1,
  and then 'make install' for installation.

Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash sbindir=/nix/store/81p3wwkrgxhhad2q60j9gb4i14ipg2iz-usbmuxd2-unstable-2023-12-12/bin
make  all-recursive
make[1]: Entering directory '/build/source'
Making all in usbmuxd2
make[2]: Entering directory '/build/source/usbmuxd2'
clang++ -DHAVE_CONFIG_H -I. -I..       -I../usbmuxd2  -I/nix/store/lfdf9z0mh5g8ihcx1pi9ak17rmncf81s-libplist-2.6.0-dev/include -I/nix/store/b3xy0gi9vlfcxcmf8yrb1ckfy4si872d-libusb-1.0.27-dev/include/libusb-1.0  -I/nix/store/qrh0njcbgr6>
clang -DHAVE_CONFIG_H -I. -I..    -I../usbmuxd2  -I/nix/store/lfdf9z0mh5g8ihcx1pi9ak17rmncf81s-libplist-2.6.0-dev/include -I/nix/store/b3xy0gi9vlfcxcmf8yrb1ckfy4si872d-libusb-1.0.27-dev/include/libusb-1.0  -I/nix/store/qrh0njcbgr6056w6>
clang++ -DHAVE_CONFIG_H -I. -I..       -I../usbmuxd2  -I/nix/store/lfdf9z0mh5g8ihcx1pi9ak17rmncf81s-libplist-2.6.0-dev/include -I/nix/store/b3xy0gi9vlfcxcmf8yrb1ckfy4si872d-libusb-1.0.27-dev/include/libusb-1.0  -I/nix/store/qrh0njcbgr6>
clang++ -DHAVE_CONFIG_H -I. -I..       -I../usbmuxd2  -I/nix/store/lfdf9z0mh5g8ihcx1pi9ak17rmncf81s-libplist-2.6.0-dev/include -I/nix/store/b3xy0gi9vlfcxcmf8yrb1ckfy4si872d-libusb-1.0.27-dev/include/libusb-1.0  -I/nix/store/qrh0njcbgr6>
Muxer.cpp:216:21: error: no matching function for call to 'find'
  216 |                     std::find(wifidev->_ipaddr.begin(), wifidev->_ipaddr.end(), nip) != wifidev->_ipaddr.end()) {
      |                     ^~~~~~~~~
/nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/streambuf_iterator.h:435:5: note: candidate template ignored: could not match 'istreambuf_iterator' against '__normal_iterator'
  435 |     find(istreambuf_iterator<_CharT> __first,
      |     ^
Muxer.cpp:261:21: error: no matching function for call to 'find'
  261 |                 if (std::find(wifidev->_ipaddr.begin(), wifidev->_ipaddr.end(), nip) != wifidev->_ipaddr.end()) {
      |                     ^~~~~~~~~
/nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/include/c++/14-20241116/bits/streambuf_iterator.h:435:5: note: candidate template ignored: could not match 'istreambuf_iterator' against '__normal_iterator'
  435 |     find(istreambuf_iterator<_CharT> __first,
      |     ^
2 errors generated.
make[2]: *** [Makefile:655: usbmuxd-Muxer.o] Error 1
make[2]: Leaving directory '/build/source/usbmuxd2'
make[1]: *** [Makefile:442: all-recursive] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:372: all] Error 2
```

</details>

## Additional context

issue is reproducible in a nixos minimal latest vm.

/nix/store/h6zgjy47g85qd5k8jgqrb3n0f8p9v39v-usbmuxd2-unstable-2023-12-12.drv
is a version that verifiably works.

https://hydra.nixos.org/build/282584321 seems to be the problematic hydra build. 

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

Main system:
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.58, NixOS, 24.11 (Vicuna), 24.11.20241025.18536bf`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - channels(root): `"nixos, nixpkgs"`
 - nixpkgs: `/nix/store/wb6agba4kfsxpbnb5hzlq58vkjzvbsk6-source`

VM:
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.67, NixOS, 25.05 (Warbler), 25.05beta729082.634fd4680144`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos-25.05beta729082.634fd4680144"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@onny 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/us/usbmuxd2/package.nix
+++ b/pkgs/by-name/us/usbmuxd2/package.nix
@@ -53,6 +53,9 @@ clangStdenv.mkDerivation rec {
   postPatch = ''
     # Checking for libgeneral version still fails
     sed -i 's/libgeneral >= $LIBGENERAL_MINVERS_STR/libgeneral/' configure.ac
+
+    # Otherwise, it will complain about no matching function for call to 'find'
+    sed -i 1i'#include <algorithm>' usbmuxd2/Muxer.cpp
   '';
 
   nativeBuildInputs = [

